ContactMechanics/Systems.py

```diff
@@ -485,6 +485,11 @@ def fun(disp):
 
         return fun
 
+    def hessp(self, disp):
```

Comment:
Can we explicit here? I guess this is the hessian product, then call this function `hessian_product`. A docstring would also be useful.